File: 6kyu/break-camelCase.js
//Describtion:

//My Solution:
function solution(string) {
    return string.replace(/([A-Z][a-z])/g,' $1').replace(/(\d)/g,' $1')
  }

 //Other solutions:
 function solution(string) {
    string = string.split('').map(function (el) {
      if (el === el.toUpperCase()) {
        el = ' ' + el
      }
      return el
    })
    return string.join('')
  }
  
//Another one:
function solution(string) {
    return(string.replace(/([A-Z])/g, ' $1'));
  
  }

File: Python/6kyu/break-camelCase.py
#Describtion:
# Complete the solution so that the function will break up camel casing, using a space between words.
# Example

# "camelCasing"  =>  "camel Casing"
# "identifier"   =>  "identifier"
# ""             =>  ""

#Solution:
def solution(s):
     string = ''
     for letter in s:
          if letter != letter.upper():
               string += letter
          if letter == letter.upper():
               string += " " + letter
          return string 
          
#Other solutions:
def solution(s):
    return ''.join(' ' + c if c.isupper() else c for c in s)

#Another:
import re
def solution(s):
    return re.sub('([A-Z])', r' \1', s)